tools/crm_resource_print.c

```diff
@@ -137,6 +137,77 @@ cli_resource_print(pcmk_resource_t *rsc, pcmk_scheduler_t *scheduler,
     return pcmk_rc_ok;
 }
 
+PCMK__OUTPUT_ARGS("attribute-changed", "const char *", "const char *",
+                  "const char *", "const char *", "const char *", "const char *",
+                  "pcmk_resource_t *")
+static int
+attribute_changed_default(pcmk__output_t *out, va_list args)
+{
+    const char *attr_set_type G_GNUC_UNUSED = va_arg(args, const char *);
+    const char *attr_set_id = va_arg(args, const char *);
+    const char *name = va_arg(args, const char *);
+    const char *value = va_arg(args, const char *);
+    const char *given_rsc_id = va_arg(args, const char *);
+    const char *found_attr_id = va_arg(args, const char *);
+    pcmk_resource_t *rsc G_GNUC_UNUSED = va_arg(args, pcmk_resource_t *);
+
+    out->info(out, "Set '%s' option: "
+              PCMK_XA_ID "=%s%s%s%s%s value=%s",
+              given_rsc_id, found_attr_id,
+              ((attr_set_id == NULL)? "" : " " PCMK__XA_SET "="),
+              pcmk__s(attr_set_id, ""),
+              ((name == NULL)? "" : " " PCMK_XA_NAME "="),
+              pcmk__s(name, ""), value);
+
+    return pcmk_rc_ok;
+}
+
+PCMK__OUTPUT_ARGS("attribute-changed", "const char *", "const char *",
+                  "const char *", "const char *", "const char *", "const char *",
+                  "pcmk_resource_t *")
+static int
+attribute_changed_xml(pcmk__output_t *out, va_list args)
+{
+    const char *attr_set_type = va_arg(args, const char *);
+    const char *attr_set_id = va_arg(args, const char *);
+    const char *name = va_arg(args, const char *);
+    const char *value = va_arg(args, const char *);
+    const char *given_rsc_id G_GNUC_UNUSED = va_arg(args, const char *);
+    const char *found_attr_id = va_arg(args, const char *);
+    pcmk_resource_t *rsc = va_arg(args, pcmk_resource_t *);
+
+    const char *variant = NULL;
+
+    if (rsc->variant == pcmk_rsc_variant_bundle) {
+        variant = PCMK_XE_BUNDLE;
+    } else if (rsc->variant == pcmk_rsc_variant_clone) {
+        variant = PCMK_XE_CLONE;
+    } else if (rsc->variant == pcmk_rsc_variant_group) {
+        variant = PCMK_XE_GROUP;
+    } else {
+        variant = PCMK_XE_PRIMITIVE;
+    }
+
+    pcmk__output_xml_create_parent(out, variant,
+                                   PCMK_XA_ID, rsc->id,
+                                   NULL);
+
+    pcmk__output_xml_create_parent(out, attr_set_type,
```

Comment:
attr_set_type can be NULL, in which `find_resource_attr()` will search for an nvpair in any block. In that case we should use xml_search->parent->name. Probably need to bubble up xml_search instead of just the attribute value.